swarm-derive/tests/test.rs

```diff
@@ -201,18 +166,27 @@ fn custom_event_no_polling() {
 fn custom_event_and_polling() {
     #[allow(dead_code)]
     #[derive(NetworkBehaviour)]
-    #[behaviour(poll_method = "foo", out_event = "String", event_process = true)]
+    #[behaviour(poll_method = "foo", out_event = "MyEvent")]
```

Comment:
Do we still need the `poll_method` config? I don't think there is a usecase now that we removed `NetworkBehaviourEventProcess`.

Reply:
I don't think we need neither `poll_method` nor `#[behaviour(ignore)]`, though I would start this discussion once this is merged. Does that sound good to you @thomaseizinger or would you prefer removing the two within this pull request?

Reply:
I think you are right.

We might also want to consider getting a release out first. With `NetworkBehaviourEventProcess` being removed, the removal of `poll_method` and `ignore` should be easy, i.e. most people will have changed their code already.

Actually, what do you think of kicking out a release before this PR to deprecate `NetworkBehaviourEventProcess`? Depending on the design of the application, changing this could be a serious effort and just removing it may block users from upgrading for a while.

Reply:
> Actually, what do you think of kicking out a release before this PR to deprecate `NetworkBehaviourEventProcess`? Depending on the design of the application, changing this could be a serious effort and just removing it may block users from upgrading for a while.

How would the two scenarios below differ in terms of user experience?

1. We deprecate `NetworkBehaviourEventProcess`, cut a release, users upgrade removing their `NetworkBehaviourEventProcess` implementations, we merge here, cut a second release, users update to the second release.
2. We merge here, cut a release, users upgrade removing their `NetworkBehaviourEventProcess` implementations.

Is (2) not just simpler than (1)? Are there any changes in `master` that users need in a timely fashion where the removal of `NetworkBehaviourEventProcess` would add a painful delay @thomaseizinger?

Reply:
The difference is that deprecation warnings can be resolved incrementally whereas compile errors have to be fixed in one go.

I don't have a strong opinion on it but I do think that deprecating first will be easier for users.

Reply:
Depending on the application, I could imagine this change to require some serious work. I think you actually once mentioned a case where a libp2p-powered app basically had its entire app logic in these callbacks.

Reply:
:+1: makes sense to me. Thanks @thomaseizinger. In addition, it might surface a use-case that we have not thought about and that might not be supported by `out_event`.

Want to take a look at https://github.com/libp2p/rust-libp2p/pull/2784?